In [ ]:













import pandas as pd
import numpy as np

In [2]:
import warnings
warnings.filterwarnings('ignore')

In [3]:
df = pd.read_csv("pip_dataset.csv")

In [4]:
df

country  year reporting_level welfare_type  ppp_version  survey_year  \
0      Albania  1996        national  consumption         2011       1996.0   
1      Albania  2002        national  consumption         2011       2002.0   
2      Albania  2005        national  consumption         2011       2005.0   
3      Albania  2008        national  consumption         2011       2008.0   
4      Albania  2012        national  consumption         2011       2012.0   
...        ...   ...             ...          ...          ...          ...   
4872    Zambia  2010        national  consumption         2017       2010.0   
4873    Zambia  2015        national  consumption         2017       2015.0   
4874  Zimbabwe  2011        national  consumption         2017       2011.0   
4875  Zimbabwe  2017        national  consumption         2017       2017.0   
4876  Zimbabwe  2019        national  consumption         2017       2019.0   

      survey_comparability  headcount_ratio_international_povline  \
0                      0.0                               0.920669   
1                      1.0                               1.570843   
2                      1.0                               0.860527   
3                      1.0                               0.313650   
4                      1.0                               0.849754   
...                    ...                                    ...   
4872                   4.0                              68.456606   
4873                   4.0                              61.352160   
4874                   0.0                              21.580063   
4875                   0.0                              34.206046   
4876                   0.0                              39.754533   

      headcount_ratio_lower_mid_income_povline  \
0                                    11.174149   
1                                    14.132118   
2                                     8.715685   
3                                     5.250542   
4                                     6.182414   
...                                        ...   
4872                                 82.885088   
4873                                 77.548045   
4874                                 47.857143   
4875                                 61.583570   
4876                                 64.509857   

      headcount_ratio_upper_mid_income_povline  ...  decile8_thr  decile9_thr  \
0                                    44.618417  ...         8.85        10.92   
1                                    49.669635  ...         8.83        11.58   
2                                    38.545254  ...        10.02        12.78   
3                                    31.110345  ...        10.74        13.62   
4                                    34.528906  ...        10.52        13.26   
...                                        ...  ...          ...          ...   
4872                                 93.044964  ...         3.27         5.34   
4873                                 90.747141  ...         4.02         6.43   
4874                                 77.945113  ...         7.25        10.54   
4875                                 84.109019  ...         5.96         8.81   
4876                                 85.016315  ...         5.76         9.01   

          gini       mld  polarization  palma_ratio  s80_s20_ratio  \
0     0.270103  0.119104      0.241293     0.928335       3.945872   
1     0.317390  0.164812      0.268982     1.215056       4.831625   
2     0.305957  0.154413      0.254529     1.142718       4.662236   
3     0.299847  0.148893      0.247311     1.114657       4.395911   
4     0.289605  0.138417      0.249988     1.041193       4.272573   
...        ...       ...           ...          ...            ...   
4872  0.556215  0.539135      0.536149     4.278696      16.024650   
4873  0.571361  0.604667      0.613181     4.995829      21.243915   
4874  0.431536  0.311153 

In [5]:
df.head()

country  year reporting_level welfare_type  ppp_version  survey_year  \
0  Albania  1996        national  consumption         2011       1996.0   
1  Albania  2002        national  consumption         2011       2002.0   
2  Albania  2005        national  consumption         2011       2005.0   
3  Albania  2008        national  consumption         2011       2008.0   
4  Albania  2012        national  consumption         2011       2012.0   

   survey_comparability  headcount_ratio_international_povline  \
0                   0.0                               0.920669   
1                   1.0                               1.570843   
2                   1.0                               0.860527   
3                   1.0                               0.313650   
4                   1.0                               0.849754   

   headcount_ratio_lower_mid_income_povline  \
0                                 11.174149   
1                                 14.132118   
2                                  8.715685   
3                                  5.250542   
4                                  6.182414   

   headcount_ratio_upper_mid_income_povline  ...  decile8_thr  decile9_thr  \
0                                 44.618417  ...         8.85        10.92   
1                                 49.669635  ...         8.83        11.58   
2                                 38.545254  ...        10.02        12.78   
3                                 31.110345  ...        10.74        13.62   
4                                 34.528906  ...        10.52        13.26   

       gini       mld  polarization  palma_ratio  s80_s20_ratio  \
0  0.270103  0.119104      0.241293     0.928335       3.945872   
1  0.317390  0.164812      0.268982     1.215056       4.831625   
2  0.305957  0.154413      0.254529     1.142718       4.662236   
3  0.299847  0.148893      0.247311     1.114657       4.395911   
4  0.289605  0.138417      0.249988     1.041193       4.272573   

   p90_p10_ratio  p90_p50_ratio  p50_p10_ratio  
0       3.568627       1.889273       1.888889  
1       3.979381       2.090253       1.903780  
2       3.872727       1.978328       1.957576  
3       3.574803       1.956897       1.826772  
4       3.632877       1.941435       1.871233  

[5 rows x 108 columns]

In [6]:
df.drop_duplicates()

country  year reporting_level welfare_type  ppp_version  survey_year  \
0      Albania  1996        national  consumption         2011       1996.0   
1      Albania  2002        national  consumption         2011       2002.0   
2      Albania  2005        national  consumption         2011       2005.0   
3      Albania  2008        national  consumption         2011       2008.0   
4      Albania  2012        national  consumption         2011       2012.0   
...        ...   ...             ...          ...          ...          ...   
4872    Zambia  2010        national  consumption         2017       2010.0   
4873    Zambia  2015        national  consumption         2017       2015.0   
4874  Zimbabwe  2011        national  consumption         2017       2011.0   
4875  Zimbabwe  2017        national  consumption         2017       2017.0   
4876  Zimbabwe  2019        national  consumption         2017       2019.0   

      survey_comparability  headcount_ratio_international_povline  \
0                      0.0                               0.920669   
1                      1.0                               1.570843   
2                      1.0                               0.860527   
3                      1.0                               0.313650   
4                      1.0                               0.849754   
...                    ...                                    ...   
4872                   4.0                              68.456606   
4873                   4.0                              61.352160   
4874                   0.0                              21.580063   
4875                   0.0                              34.206046   
4876                   0.0                              39.754533   

      headcount_ratio_lower_mid_income_povline  \
0                                    11.174149   
1                                    14.132118   
2                                     8.715685   
3                                     5.250542   
4                                     6.182414   
...                                        ...   
4872                                 82.885088   
4873                                 77.548045   
4874                                 47.857143   
4875                                 61.583570   
4876                                 64.509857   

      headcount_ratio_upper_mid_income_povline  ...  decile8_thr  decile9_thr  \
0                                    44.618417  ...         8.85        10.92   
1                                    49.669635  ...         8.83        11.58   
2                                    38.545254  ...        10.02        12.78   
3                                    31.110345  ...        10.74        13.62   
4                                    34.528906  ...        10.52        13.26   
...                                        ...  ...          ...          ...   
4872                                 93.044964  ...         3.27         5.34   
4873                                 90.747141  ...         4.02         6.43   
4874                                 77.945113  ...         7.25        10.54   
4875                                 84.109019  ...         5.96         8.81   
4876                                 85.016315  ...         5.76         9.01   

          gini       mld  polarization  palma_ratio  s80_s20_ratio  \
0     0.270103  0.119104      0.241293     0.928335       3.945872   
1     0.317390  0.164812      0.268982     1.215056       4.831625   
2     0.305957  0.154413      0.254529     1.142718       4.662236   
3     0.299847  0.148893      0.247311     1.114657       4.395911   
4     0.289605  0.138417      0.249988     1.041193       4.272573   
...        ...       ...           ...          ...            ...   
4872  0.556215  0.539135      0.536149     4.278696      16.024650   
4873  0.571361  0.604667      0.613181     4.995829      21.243915   
4874  0.431536  0.311153 

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4877 entries, 0 to 4876
Columns: 108 entries, country to p50_p10_ratio
dtypes: float64(103), int64(2), object(3)
memory usage: 4.0+ MB


In [8]:
df.describe()

year  ppp_version  survey_year  survey_comparability  \
count  4877.000000  4877.000000  4411.000000           4411.000000   
mean   2005.759893  2013.999385  2005.939896              1.639084   
std       9.438782     3.000308     9.497211              1.402898   
min    1967.000000  2011.000000  1967.000000              0.000000   
25%    2000.000000  2011.000000  2000.000000              1.000000   
50%    2007.000000  2011.000000  2007.000000              1.000000   
75%    2013.000000  2017.000000  2014.000000              2.000000   
max    2021.000000  2017.000000  2021.000000              6.000000   

       headcount_ratio_international_povline  \
count                            4877.000000   
mean                               11.081565   
std                                18.190433   
min                                 0.000000   
25%                                 0.278387   
50%                                 2.041412   
75%                                13.390963   
max                                96.871427   

       headcount_ratio_lower_mid_income_povline  \
count                               4877.000000   
mean                                  21.672687   
std                                   27.104145   
min                                    0.000000   
25%                                    0.807877   
50%                                    9.186717   
75%                                   34.166712   
max                                   99.999000   

       headcount_ratio_upper_mid_income_povline  headcount_ratio_100  \
count                               4877.000000          4877.000000   
mean                                  36.538818             3.164721   
std                                   33.674140             7.568626   
min                                    0.000000             0.000000   
25%                                    3.032081             0.044772   
50%                                   28.582685             0.371755   
75%                                   64.864719             2.158480   
max                                   99.999000            79.532619   

       headcount_ratio_1000  headcount_ratio_2000  ...  decile8_thr  \
count           4877.000000           4877.000000  ...  4877.000000   
mean              50.351520             69.105964  ...    26.273736   
std               36.002943             34.132391  ...    24.171141   
min                0.000000              0.920154  ...     1.010000   
25%               10.328433             44.575897  ...     8.680000   
50%               55.337658             84.916259  ...    16.980000   
75%               85.133887             97.179433  ...    35.550000   
max               99.999000            100.000000  ...   120.100000   

       decile9_thr         gini          mld  polarization  palma_ratio  \
count  4877.000000  4401.000000  4401.000000   4401.000000  4395.000000   
mean     35.110607     0.375645     0.264371      0.327385     1.886528   
std      30.671393     0.088840     0.140325      0.101136     1.138502   
min       1.460000     0.177920     0.053563      0.146643     0.596408   
25%      11.960000     0.308719     0.163090      0.252448     1.154071   
50%      24.560000     0.355622     0.220968      0.300364     1.465800   
75%      48.350000     0.427676     0.320859      0.380201     2.152142   
max     164.700000     0.657556     0.937047      0.815704     8.343586   

       s80_s20_ratio  p90_p10_ratio  p90_p50_ratio  p50_p10_ratio  
count    4395.000000    4877.000000    4877.000000    4877.000000  
mean        8.219764       7.424451       2.532256       2.660335  
std         6.018946      42.891943       0.954387      11.785204  
min         2.430173       2.191176       1.475248       1.485294  
25%         4.720943       3.937500       2.017751       1.942113  
50%         6.171947       5.149826       2.290076       2.207904  
75%         9.043292       7.458333       2.78091

In [9]:
df.columns

Index(['country', 'year', 'reporting_level', 'welfare_type', 'ppp_version',
       'survey_year', 'survey_comparability',
       'headcount_ratio_international_povline',
       'headcount_ratio_lower_mid_income_povline',
       'headcount_ratio_upper_mid_income_povline',
       ...
       'decile8_thr', 'decile9_thr', 'gini', 'mld', 'polarization',
       'palma_ratio', 's80_s20_ratio', 'p90_p10_ratio', 'p90_p50_ratio',
       'p50_p10_ratio'],
      dtype='object', length=108)

In [10]:
df.isnull()

country   year  reporting_level  welfare_type  ppp_version  survey_year  \
0       False  False            False         False        False        False   
1       False  False            False         False        False        False   
2       False  False            False         False        False        False   
3       False  False            False         False        False        False   
4       False  False            False         False        False        False   
...       ...    ...              ...           ...          ...          ...   
4872    False  False            False         False        False        False   
4873    False  False            False         False        False        False   
4874    False  False            False         False        False        False   
4875    False  False            False         False        False        False   
4876    False  False            False         False        False        False   

      survey_comparability  headcount_ratio_international_povline  \
0                    False                                  False   
1                    False                                  False   
2                    False                                  False   
3                    False                                  False   
4                    False                                  False   
...                    ...                                    ...   
4872                 False                                  False   
4873                 False                                  False   
4874                 False                                  False   
4875                 False                                  False   
4876                 False                                  False   

      headcount_ratio_lower_mid_income_povline  \
0                                        False   
1                                        False   
2                                        False   
3                                        False   
4                                        False   
...                                        ...   
4872                                     False   
4873                                     False   
4874                                     False   
4875                                     False   
4876                                     False   

      headcount_ratio_upper_mid_income_povline  ...  decile8_thr  decile9_thr  \
0                                        False  ...        False        False   
1                                        False  ...        False        False   
2                                        False  ...        False        False   
3                                        False  ...        False        False   
4                                        False  ...        False        False   
...                                        ...  ...          ...          ...   
4872                                     False  ...        False        False   
4873                                     False  ...        False        False   
4874                                     False  ...        False        False   
4875                                     False  ...        False        False   
4876                                     False  ...        False        False   

       gini    mld  polarization  palma_ratio  s80_s20_ratio  p90_p10_ratio  \
0     False  False         False        False          False          False   
1     False  False         False        False          False          False   
2     False  False         False        False          False          False   
3     False  False         False        False          False          False   
4     False  False         False        False          False          False   
...     ...    ...           ...          ...            ...            ...   
4872  False  False         False        False          False          False  

In [11]:
missing_values_per_column = df.isnull().sum()
print(missing_values_per_column)


country              0
year                 0
reporting_level    466
welfare_type       466
ppp_version          0
                  ... 
palma_ratio        482
s80_s20_ratio      482
p90_p10_ratio        0
p90_p50_ratio        0
p50_p10_ratio        0
Length: 108, dtype: int64


In [12]:
# Check for missing values and their counts
missing_values = df.isnull().sum()

# Drop columns with a high percentage of missing values (e.g., more than 50% missing)
threshold = 0.5 * len(df)
data_cleaned = df.dropna(thresh=threshold, axis=1)

# Recheck missing values after dropping
remaining_missing_values = data_cleaned.isnull().sum()

# Convert 'year' and 'survey_year' to integer for consistency (fill missing with mode or median)
data_cleaned['year'] = data_cleaned['year'].fillna(data_cleaned['year'].median()).astype(int)
data_cleaned['survey_year'] = data_cleaned['survey_year'].fillna(data_cleaned['survey_year'].median()).astype(int)

# Check column names for possible renaming
columns_before_rename = data_cleaned.columns

# Output cleaning summary and the columns before renaming
missing_values, remaining_missing_values, columns_before_rename


(country              0
 year                 0
 reporting_level    466
 welfare_type       466
 ppp_version          0
                   ... 
 palma_ratio        482
 s80_s20_ratio      482
 p90_p10_ratio        0
 p90_p50_ratio        0
 p50_p10_ratio        0
 Length: 108, dtype: int64,
 country              0
 year                 0
 reporting_level    466
 welfare_type       466
 ppp_version          0
                   ... 
 palma_ratio        482
 s80_s20_ratio      482
 p90_p10_ratio        0
 p90_p50_ratio        0
 p50_p10_ratio        0
 Length: 108, dtype: int64,
 Index(['country', 'year', 'reporting_level', 'welfare_type', 'ppp_version',
        'survey_year', 'survey_comparability',
        'headcount_ratio_international_povline',
        'headcount_ratio_lower_mid_income_povline',
        'headcount_ratio_upper_mid_income_povline',
        ...
        'decile8_thr', 'decile9_thr', 'gini', 'mld', 'polarization',
        'palma_ratio', 's80_s20_ratio', 'p90_p10_ratio', '

In [13]:
# Renaming columns for clarity
data_cleaned_renamed = data_cleaned.rename(columns={
    'headcount_ratio_international_povline': 'hc_ratio',
    'gini': 'gini_index',
    'mld': 'mean_log_deviation',
    'palma_ratio': 'palma_index',
    's80_s20_ratio': 's80_s20_index',
    'p90_p10_ratio': 'p90_p10_index',
    'p90_p50_ratio': 'p90_p50_index',
    'p50_p10_ratio': 'p50_p10_index',
    'poverty_gap_index_international_povline':'international_poverty_gap',
    'poverty_gap_index_100':'$1_poverty_gap'
})

# Dropping columns with more than 50% missing values if requested by user (already kept those with fewer missing values)
data_final = data_cleaned_renamed.dropna(subset=['reporting_level', 'welfare_type'])

# Checking the final cleaned dataframe's structure
cleaned_columns = data_final.columns
final_info = data_final.info()

cleaned_columns, final_info

<class 'pandas.core.frame.DataFrame'>
Index: 4411 entries, 0 to 4876
Columns: 108 entries, country to p50_p10_index
dtypes: float64(102), int64(3), object(3)
memory usage: 3.7+ MB


(Index(['country', 'year', 'reporting_level', 'welfare_type', 'ppp_version',
        'survey_year', 'survey_comparability', 'hc_ratio',
        'headcount_ratio_lower_mid_income_povline',
        'headcount_ratio_upper_mid_income_povline',
        ...
        'decile8_thr', 'decile9_thr', 'gini_index', 'mean_log_deviation',
        'polarization', 'palma_index', 's80_s20_index', 'p90_p10_index',
        'p90_p50_index', 'p50_p10_index'],
       dtype='object', length=108),
 None)

In [14]:
data_final

country  year reporting_level welfare_type  ppp_version  survey_year  \
0      Albania  1996        national  consumption         2011         1996   
1      Albania  2002        national  consumption         2011         2002   
2      Albania  2005        national  consumption         2011         2005   
3      Albania  2008        national  consumption         2011         2008   
4      Albania  2012        national  consumption         2011         2012   
...        ...   ...             ...          ...          ...          ...   
4872    Zambia  2010        national  consumption         2017         2010   
4873    Zambia  2015        national  consumption         2017         2015   
4874  Zimbabwe  2011        national  consumption         2017         2011   
4875  Zimbabwe  2017        national  consumption         2017         2017   
4876  Zimbabwe  2019        national  consumption         2017         2019   

      survey_comparability   hc_ratio  \
0                      0.0   0.920669   
1                      1.0   1.570843   
2                      1.0   0.860527   
3                      1.0   0.313650   
4                      1.0   0.849754   
...                    ...        ...   
4872                   4.0  68.456606   
4873                   4.0  61.352160   
4874                   0.0  21.580063   
4875                   0.0  34.206046   
4876                   0.0  39.754533   

      headcount_ratio_lower_mid_income_povline  \
0                                    11.174149   
1                                    14.132118   
2                                     8.715685   
3                                     5.250542   
4                                     6.182414   
...                                        ...   
4872                                 82.885088   
4873                                 77.548045   
4874                                 47.857143   
4875                                 61.583570   
4876                                 64.509857   

      headcount_ratio_upper_mid_income_povline  ...  decile8_thr  decile9_thr  \
0                                    44.618417  ...         8.85        10.92   
1                                    49.669635  ...         8.83        11.58   
2                                    38.545254  ...        10.02        12.78   
3                                    31.110345  ...        10.74        13.62   
4                                    34.528906  ...        10.52        13.26   
...                                        ...  ...          ...          ...   
4872                                 93.044964  ...         3.27         5.34   
4873                                 90.747141  ...         4.02         6.43   
4874                                 77.945113  ...         7.25        10.54   
4875                                 84.109019  ...         5.96         8.81   
4876                                 85.016315  ...         5.76         9.01   

      gini_index  mean_log_deviation  polarization  palma_index  \
0       0.270103            0.119104      0.241293     0.928335   
1       0.317390            0.164812      0.268982     1.215056   
2       0.305957            0.154413      0.254529     1.142718   
3       0.299847            0.148893      0.247311     1.114657   
4       0.289605            0.138417      0.249988     1.041193   
...          ...                 ...           ...          ...   
4872    0.556215            0.539135      0.536149     4.278696   
4873    0.571361            0.604667      0.613181     4.995829   
4874    0.431536            0.311153      0.388356     2.207938   
4875    0.443371            0.322848      0.416694     2.307359   
4876    0.502565            0.427862      0.463899          NaN   

      s80_s20_index  p90_p10_index  p90_p50_index  p50_p10_index  
0          3.945872       3.568627       1.889273       1.888889  
1          4.831625       3.979381       2.090253   

In [15]:
cleaned_columns

Index(['country', 'year', 'reporting_level', 'welfare_type', 'ppp_version',
       'survey_year', 'survey_comparability', 'hc_ratio',
       'headcount_ratio_lower_mid_income_povline',
       'headcount_ratio_upper_mid_income_povline',
       ...
       'decile8_thr', 'decile9_thr', 'gini_index', 'mean_log_deviation',
       'polarization', 'palma_index', 's80_s20_index', 'p90_p10_index',
       'p90_p50_index', 'p50_p10_index'],
      dtype='object', length=108)

In [16]:
# Drop duplicate rows
data_final_no_duplicates = data_final.drop_duplicates()

# Standardize categorical variables (lowercase and strip whitespace)
data_final_no_duplicates['country'] = data_final_no_duplicates['country'].str.lower().str.strip()
data_final_no_duplicates['reporting_level'] = data_final_no_duplicates['reporting_level'].str.lower().str.strip()
data_final_no_duplicates['welfare_type'] = data_final_no_duplicates['welfare_type'].str.lower().str.strip()

# Check the final cleaned dataframe's structure and any duplicates
final_no_duplicates_info = data_final_no_duplicates.info()
duplicates_count = data_final.shape[0] - data_final_no_duplicates.shape[0]

final_no_duplicates_info, duplicates_count


<class 'pandas.core.frame.DataFrame'>
Index: 4411 entries, 0 to 4876
Columns: 108 entries, country to p50_p10_index
dtypes: float64(102), int64(3), object(3)
memory usage: 3.7+ MB


(None, 0)

In [17]:
data_final_no_duplicates

country  year reporting_level welfare_type  ppp_version  survey_year  \
0      albania  1996        national  consumption         2011         1996   
1      albania  2002        national  consumption         2011         2002   
2      albania  2005        national  consumption         2011         2005   
3      albania  2008        national  consumption         2011         2008   
4      albania  2012        national  consumption         2011         2012   
...        ...   ...             ...          ...          ...          ...   
4872    zambia  2010        national  consumption         2017         2010   
4873    zambia  2015        national  consumption         2017         2015   
4874  zimbabwe  2011        national  consumption         2017         2011   
4875  zimbabwe  2017        national  consumption         2017         2017   
4876  zimbabwe  2019        national  consumption         2017         2019   

      survey_comparability   hc_ratio  \
0                      0.0   0.920669   
1                      1.0   1.570843   
2                      1.0   0.860527   
3                      1.0   0.313650   
4                      1.0   0.849754   
...                    ...        ...   
4872                   4.0  68.456606   
4873                   4.0  61.352160   
4874                   0.0  21.580063   
4875                   0.0  34.206046   
4876                   0.0  39.754533   

      headcount_ratio_lower_mid_income_povline  \
0                                    11.174149   
1                                    14.132118   
2                                     8.715685   
3                                     5.250542   
4                                     6.182414   
...                                        ...   
4872                                 82.885088   
4873                                 77.548045   
4874                                 47.857143   
4875                                 61.583570   
4876                                 64.509857   

      headcount_ratio_upper_mid_income_povline  ...  decile8_thr  decile9_thr  \
0                                    44.618417  ...         8.85        10.92   
1                                    49.669635  ...         8.83        11.58   
2                                    38.545254  ...        10.02        12.78   
3                                    31.110345  ...        10.74        13.62   
4                                    34.528906  ...        10.52        13.26   
...                                        ...  ...          ...          ...   
4872                                 93.044964  ...         3.27         5.34   
4873                                 90.747141  ...         4.02         6.43   
4874                                 77.945113  ...         7.25        10.54   
4875                                 84.109019  ...         5.96         8.81   
4876                                 85.016315  ...         5.76         9.01   

      gini_index  mean_log_deviation  polarization  palma_index  \
0       0.270103            0.119104      0.241293     0.928335   
1       0.317390            0.164812      0.268982     1.215056   
2       0.305957            0.154413      0.254529     1.142718   
3       0.299847            0.148893      0.247311     1.114657   
4       0.289605            0.138417      0.249988     1.041193   
...          ...                 ...           ...          ...   
4872    0.556215            0.539135      0.536149     4.278696   
4873    0.571361            0.604667      0.613181     4.995829   
4874    0.431536            0.311153      0.388356     2.207938   
4875    0.443371            0.322848      0.416694     2.307359   
4876    0.502565            0.427862      0.463899          NaN   

      s80_s20_index  p90_p10_index  p90_p50_index  p50_p10_index  
0          3.945872       3.568627       1.889273       1.888889  
1          4.831625       3.979381       2.090253   

In [18]:
data_final.dropna(inplace=True)

<details>
<summary><strong>Problem Statement </strong></summary>
<strong>How the poverty is changing in different countries and how using this data and how it will change further</strong>
</details>

# Columnn Filtering According to the goal

In [19]:
data_final.info

<bound method DataFrame.info of        country  year reporting_level welfare_type  ppp_version  survey_year  \
0      Albania  1996        national  consumption         2011         1996   
1      Albania  2002        national  consumption         2011         2002   
2      Albania  2005        national  consumption         2011         2005   
4      Albania  2012        national  consumption         2011         2012   
8      Albania  2016        national       income         2011         2016   
...        ...   ...             ...          ...          ...          ...   
4871    Zambia  2006        national  consumption         2017         2006   
4872    Zambia  2010        national  consumption         2017         2010   
4873    Zambia  2015        national  consumption         2017         2015   
4874  Zimbabwe  2011        national  consumption         2017         2011   
4875  Zimbabwe  2017        national  consumption         2017         2017   

      survey_compar

In [20]:
data_final.describe()

year  ppp_version  survey_year  survey_comparability  \
count  3643.000000  3643.000000  3643.000000           3643.000000   
mean   2005.715893  2013.969531  2005.715344              1.701071   
std       9.498677     3.000257     9.497996              1.434786   
min    1967.000000  2011.000000  1967.000000              0.000000   
25%    2000.000000  2011.000000  2000.000000              1.000000   
50%    2007.000000  2011.000000  2007.000000              1.000000   
75%    2013.000000  2017.000000  2013.000000              3.000000   
max    2021.000000  2017.000000  2021.000000              6.000000   

          hc_ratio  headcount_ratio_lower_mid_income_povline  \
count  3643.000000                               3643.000000   
mean     12.138591                                 23.630968   
std      18.918983                                 27.741979   
min       0.003111                                  0.008470   
25%       0.499448                                  1.220419   
50%       3.040325                                 11.449953   
75%      14.733494                                 37.948199   
max      96.871427                                 99.999000   

       headcount_ratio_upper_mid_income_povline  headcount_ratio_100  \
count                               3643.000000          3643.000000   
mean                                  39.082788             3.517321   
std                                   34.268113             8.060688   
min                                    0.027518             0.000002   
25%                                    3.548506             0.126501   
50%                                   33.057065             0.499486   
75%                                   69.792769             2.678522   
max                                   99.999000            79.532619   

       headcount_ratio_1000  headcount_ratio_2000  ...  decile8_thr  \
count           3643.000000           3643.000000  ...  3643.000000   
mean              52.326520             69.721990  ...    25.817812   
std               36.609965             34.111799  ...    24.345944   
min                0.041838              0.920154  ...     1.010000   
25%               11.463004             44.979920  ...     7.790000   
50%               58.637482             86.124638  ...    16.320000   
75%               88.401326             98.072528  ...    35.575000   
max               99.999000            100.000000  ...   120.100000   

       decile9_thr   gini_index  mean_log_deviation  polarization  \
count  3643.000000  3643.000000         3643.000000   3643.000000   
mean     34.455040     0.388367            0.283644      0.340420   
std      30.872668     0.089323            0.143887      0.103325   
min       1.460000     0.177920            0.053563      0.146643   
25%      10.660000     0.321284            0.178839      0.265295   
50%      23.760000     0.369068            0.240375      0.316390   
75%      48.875000     0.447705            0.349448      0.399454   
max     164.700000     0.657556            0.937047      0.815704   

       palma_index  s80_s20_index  p90_p10_index  p90_p50_index  p50_p10_index  
count  3643.000000    3643.000000    3643.000000    3643.000000    3643.000000  
mean      2.019541       8.863798       7.520145       2.494139       2.749657  
std       1.192297       6.369985      48.333975       0.636277      13.481811  
min       0.596408       2.430173       2.191176       1.475248       1.485294  
25%       1.233440       5.059501       4.074518       2.043106       1.977706  
50%       1.567601       6.694263       5.286275       2.318966       2.250000  
75%       2.407819      10.149343       7.653601       2.811434       2.788641  
max       8.343586      72.681737    2892.000000       6.716867     809.000000  

[8 rows x 105 columns]

It means that all are numerical value which is good except 3 columns which are -:

In [21]:
data_final[ data_final.columns.difference(data_final.describe().columns)]

country reporting_level welfare_type
0      Albania        national  consumption
1      Albania        national  consumption
2      Albania        national  consumption
4      Albania        national  consumption
8      Albania        national       income
...        ...             ...          ...
4871    Zambia        national  consumption
4872    Zambia        national  consumption
4873    Zambia        national  consumption
4874  Zimbabwe        national  consumption
4875  Zimbabwe        national  consumption

[3643 rows x 3 columns]

In [22]:
data_final[['country','reporting_level','welfare_type']].nunique()

country            163
reporting_level      3
welfare_type         2
dtype: int64

we have 163 countries here and three types of reporting levels and two welfare types

In [23]:
data_final[['year','survey_year']].nunique()

year           50
survey_year    50
dtype: int64

In [24]:
data_final

country  year reporting_level welfare_type  ppp_version  survey_year  \
0      Albania  1996        national  consumption         2011         1996   
1      Albania  2002        national  consumption         2011         2002   
2      Albania  2005        national  consumption         2011         2005   
4      Albania  2012        national  consumption         2011         2012   
8      Albania  2016        national       income         2011         2016   
...        ...   ...             ...          ...          ...          ...   
4871    Zambia  2006        national  consumption         2017         2006   
4872    Zambia  2010        national  consumption         2017         2010   
4873    Zambia  2015        national  consumption         2017         2015   
4874  Zimbabwe  2011        national  consumption         2017         2011   
4875  Zimbabwe  2017        national  consumption         2017         2017   

      survey_comparability   hc_ratio  \
0                      0.0   0.920669   
1                      1.0   1.570843   
2                      1.0   0.860527   
4                      1.0   0.849754   
8                      4.0   6.681604   
...                    ...        ...   
4871                   3.0  64.698028   
4872                   4.0  68.456606   
4873                   4.0  61.352160   
4874                   0.0  21.580063   
4875                   0.0  34.206046   

      headcount_ratio_lower_mid_income_povline  \
0                                    11.174149   
1                                    14.132118   
2                                     8.715685   
4                                     6.182414   
8                                    17.621649   
...                                        ...   
4871                                 81.165271   
4872                                 82.885088   
4873                                 77.548045   
4874                                 47.857143   
4875                                 61.583570   

      headcount_ratio_upper_mid_income_povline  ...  decile8_thr  decile9_thr  \
0                                    44.618417  ...         8.85        10.92   
1                                    49.669635  ...         8.83        11.58   
2                                    38.545254  ...        10.02        12.78   
4                                    34.528906  ...        10.52        13.26   
8                                    39.949891  ...        11.88        16.02   
...                                        ...  ...          ...          ...   
4871                                 92.573347  ...         3.49         5.63   
4872                                 93.044964  ...         3.27         5.34   
4873                                 90.747141  ...         4.02         6.43   
4874                                 77.945113  ...         7.25        10.54   
4875                                 84.109019  ...         5.96         8.81   

      gini_index  mean_log_deviation  polarization  palma_index  \
0       0.270103            0.119104      0.241293     0.928335   
1       0.317390            0.164812      0.268982     1.215056   
2       0.305957            0.154413      0.254529     1.142718   
4       0.289605            0.138417      0.249988     1.041193   
8       0.385656            0.273740      0.350360     1.696002   
...          ...                 ...           ...          ...   
4871    0.546175            0.528909      0.548430     4.177633   
4872    0.556215            0.539135      0.536149     4.278696   
4873    0.571361            0.604667      0.613181     4.995829   
4874    0.431536            0.311153      0.388356     2.207938   
4875    0.443371            0.322848      0.416694     2.307359   

      s80_s20_index  p90_p10_index  p90_p50_index  p50_p10_index  
0          3.945872       3.568627       1.889273       1.888889  
1          4.831625       3.979381       2.090253   

In [25]:
keep_cols=['headcount_international_povline',
    'total_shortfall_international_povline',
    'income_gap_ratio_international_povline',
 'decile1_avg',
 'decile2_avg',
 'decile3_avg',
 'poverty_gap_index_international_povline',
 'poverty_gap_index_100',
]
headcount=[col for col in data_final if "headcount" in col]
headcount

['headcount_ratio_lower_mid_income_povline',
 'headcount_ratio_upper_mid_income_povline',
 'headcount_ratio_100',
 'headcount_ratio_1000',
 'headcount_ratio_2000',
 'headcount_ratio_3000',
 'headcount_ratio_4000',
 'headcount_ratio_40_median',
 'headcount_ratio_50_median',
 'headcount_ratio_60_median',
 'headcount_international_povline',
 'headcount_lower_mid_income_povline',
 'headcount_upper_mid_income_povline',
 'headcount_100',
 'headcount_1000',
 'headcount_2000',
 'headcount_3000',
 'headcount_4000',
 'headcount_40_median',
 'headcount_50_median',
 'headcount_60_median']

In [26]:
totalShort=[col for col in data_final if "total_shortfall" in col]
totalShort

['total_shortfall_international_povline',
 'total_shortfall_lower_mid_income_povline',
 'total_shortfall_upper_mid_income_povline',
 'total_shortfall_100',
 'total_shortfall_1000',
 'total_shortfall_2000',
 'total_shortfall_3000',
 'total_shortfall_4000',
 'total_shortfall_40_median',
 'total_shortfall_50_median',
 'total_shortfall_60_median']

In [27]:
IncomeGap=[col for col in data_final if "income_gap_ratio" in col]
IncomeGap

['income_gap_ratio_international_povline',
 'income_gap_ratio_lower_mid_income_povline',
 'income_gap_ratio_upper_mid_income_povline',
 'income_gap_ratio_100',
 'income_gap_ratio_1000',
 'income_gap_ratio_2000',
 'income_gap_ratio_3000',
 'income_gap_ratio_4000',
 'income_gap_ratio_40_median',
 'income_gap_ratio_50_median',
 'income_gap_ratio_60_median']

In [28]:
dec=[col for col in data_final if "decile" in col]
dec

['decile1_avg',
 'decile2_avg',
 'decile3_avg',
 'decile4_avg',
 'decile5_avg',
 'decile6_avg',
 'decile7_avg',
 'decile8_avg',
 'decile9_avg',
 'decile10_avg',
 'decile1_share',
 'decile2_share',
 'decile3_share',
 'decile4_share',
 'decile5_share',
 'decile6_share',
 'decile7_share',
 'decile8_share',
 'decile9_share',
 'decile10_share',
 'decile1_thr',
 'decile2_thr',
 'decile3_thr',
 'decile4_thr',
 'decile6_thr',
 'decile7_thr',
 'decile8_thr',
 'decile9_thr']

In [29]:
avg=[col for col in data_final if "avg_shortfall_" in col]
avg

['avg_shortfall_international_povline',
 'avg_shortfall_lower_mid_income_povline',
 'avg_shortfall_upper_mid_income_povline',
 'avg_shortfall_100',
 'avg_shortfall_1000',
 'avg_shortfall_2000',
 'avg_shortfall_3000',
 'avg_shortfall_4000',
 'avg_shortfall_40_median',
 'avg_shortfall_50_median',
 'avg_shortfall_60_median']

In [30]:
pgi=[col for col in data_final if "poverty_gap_index" in col]
pgi

['poverty_gap_index_lower_mid_income_povline',
 'poverty_gap_index_upper_mid_income_povline',
 'poverty_gap_index_1000',
 'poverty_gap_index_2000',
 'poverty_gap_index_3000',
 'poverty_gap_index_4000']

In [31]:
columns_to_drop = list(set(IncomeGap + dec + headcount + totalShort+avg+pgi)-set(keep_cols))+[
                                                                                            'survey_year',
                                                                                          'ppp_version',
                                                                                          'survey_comparability',
                                                                                          'welfare_type','mean',
                                                                                          'mean_log_deviation',
                                                                                          'median',
                                                                                          'p50_p10_index',
                                                                                          'p90_p10_index',
                                                                                          'p90_p50_index',
                                                                                          's80_s20_index'
                                                                                          ]
data_final.drop(columns=columns_to_drop,inplace=True)
data_final

country  year reporting_level   hc_ratio  \
0      Albania  1996        national   0.920669   
1      Albania  2002        national   1.570843   
2      Albania  2005        national   0.860527   
4      Albania  2012        national   0.849754   
8      Albania  2016        national   6.681604   
...        ...   ...             ...        ...   
4871    Zambia  2006        national  64.698028   
4872    Zambia  2010        national  68.456606   
4873    Zambia  2015        national  61.352160   
4874  Zimbabwe  2011        national  21.580063   
4875  Zimbabwe  2017        national  34.206046   

      headcount_international_povline  total_shortfall_international_povline  \
0                             29167.0                           8.430020e+03   
1                             47927.0                           1.517408e+04   
2                             25915.0                           7.514215e+03   
4                             24646.0                           9.318746e+03   
8                            192170.0                           1.347542e+05   
...                               ...                                    ...   
4871                        7876026.0                           8.804439e+06   
4872                        9314196.0                           1.035156e+07   
4873                        9742337.0                           1.118811e+07   
4874                        2782603.0                           1.452058e+06   
4875                        4869778.0                           2.871835e+06   

      income_gap_ratio_international_povline  international_poverty_gap  \
0                                  15.211893                   0.140051   
1                                  16.663586                   0.261761   
2                                  15.260853                   0.131325   
4                                  19.900199                   0.169101   
8                                  36.906527                   2.465952   
...                                      ...                        ...   
4871                               51.994342                  33.639313   
4872                               51.691832                  35.386473   
4873                               53.413998                  32.770643   
4874                               24.271370                   5.237777   
4875                               27.429122                   9.382419   

      $1_poverty_gap  decile1_avg  decile2_avg  decile3_avg  gini_index  \
0           0.011726     2.538496     3.475535     4.191319    0.270103   
1           0.024816     2.346511     3.263519     3.923426    0.317390   
2           0.000230     2.643917     3.735312     4.537911    0.305957   
4           0.011959     2.884756     4.093648     4.842953    0.289605   
8           0.732899     1.515200     2.871317     3.950169    0.385656   
...              ...          ...          ...          ...         ...   
4871       12.755820     0.356971     0.592891     0.793114    0.546175   
4872       12.591384     0.389228     0.604047     0.786941    0.556215   
4873       13.713155     0.306863     0.561880     0.781949    0.571361   
4874        0.020411     1.352072     1.828024     2.324548    0.431536   
4875        0.253692     1.118989     1.510812     1.834687    0.443371   

      polarization  palma_index  
0         0.241293     0.928335  
1         0.268982     1.215056  
2         0.254529     1.142718  
4         0.249988     1.041193  
8         0.350360     1.696002  
...            ...          ...  
4871      0.548430     4.177633  
4872      0.536149     4.278696  
4873      0.613181     4.995829  
4874      0.388356     2.207938  
4875      0.416694     2.307359  

[3643 rows x 15 columns]

In [32]:
check=[col for col in data_final.columns if col in keep_cols]
check

['headcount_international_povline',
 'total_shortfall_international_povline',
 'income_gap_ratio_international_povline',
 'decile1_avg',
 'decile2_avg',
 'decile3_avg']

In [33]:
reporting_level=data_final['reporting_level'].unique()
print(f'reporting levels are {reporting_level}')

reporting levels are ['national' 'urban' 'rural']


In [34]:
NumVal=data_final.select_dtypes(int and float)
FilteredData=data_final
#FilteredData=data_final.pivot_table(index=['country','year'],values=NumVal,columns='reporting_level')
FilteredData.reset_index()


index   country  year reporting_level   hc_ratio  \
0         0   Albania  1996        national   0.920669   
1         1   Albania  2002        national   1.570843   
2         2   Albania  2005        national   0.860527   
3         4   Albania  2012        national   0.849754   
4         8   Albania  2016        national   6.681604   
...     ...       ...   ...             ...        ...   
3638   4871    Zambia  2006        national  64.698028   
3639   4872    Zambia  2010        national  68.456606   
3640   4873    Zambia  2015        national  61.352160   
3641   4874  Zimbabwe  2011        national  21.580063   
3642   4875  Zimbabwe  2017        national  34.206046   

      headcount_international_povline  total_shortfall_international_povline  \
0                             29167.0                           8.430020e+03   
1                             47927.0                           1.517408e+04   
2                             25915.0                           7.514215e+03   
3                             24646.0                           9.318746e+03   
4                            192170.0                           1.347542e+05   
...                               ...                                    ...   
3638                        7876026.0                           8.804439e+06   
3639                        9314196.0                           1.035156e+07   
3640                        9742337.0                           1.118811e+07   
3641                        2782603.0                           1.452058e+06   
3642                        4869778.0                           2.871835e+06   

      income_gap_ratio_international_povline  international_poverty_gap  \
0                                  15.211893                   0.140051   
1                                  16.663586                   0.261761   
2                                  15.260853                   0.131325   
3                                  19.900199                   0.169101   
4                                  36.906527                   2.465952   
...                                      ...                        ...   
3638                               51.994342                  33.639313   
3639                               51.691832                  35.386473   
3640                               53.413998                  32.770643   
3641                               24.271370                   5.237777   
3642                               27.429122                   9.382419   

      $1_poverty_gap  decile1_avg  decile2_avg  decile3_avg  gini_index  \
0           0.011726     2.538496     3.475535     4.191319    0.270103   
1           0.024816     2.346511     3.263519     3.923426    0.317390   
2           0.000230     2.643917     3.735312     4.537911    0.305957   
3           0.011959     2.884756     4.093648     4.842953    0.289605   
4           0.732899     1.515200     2.871317     3.950169    0.385656   
...              ...          ...          ...          ...         ...   
3638       12.755820     0.356971     0.592891     0.793114    0.546175   
3639       12.591384     0.389228     0.604047     0.786941    0.556215   
3640       13.713155     0.306863     0.561880     0.781949    0.571361   
3641        0.020411     1.352072     1.828024     2.324548    0.431536   
3642        0.253692     1.118989     1.510812     1.834687    0.443371   

      polarization  palma_index  
0         0.241293     0.928335  
1         0.268982     1.215056  
2         0.254529     1.142718  
3         0.249988     1.041193  
4         0.350360     1.696002  
...            ...          ...  
3638      0.548430     4.177633  
3639      0.536149     4.278696  
3640      0.613181     4.995829  
3641      0.388356     2.207938  
3642      0.416694     2.307359  

[3643 rows x 16 columns]

In [35]:
FilteredData.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3643 entries, 0 to 4875
Data columns (total 15 columns):
 #   Column                                  Non-Null Count  Dtype  
---  ------                                  --------------  -----  
 0   country                                 3643 non-null   object 
 1   year                                    3643 non-null   int64  
 2   reporting_level                         3643 non-null   object 
 3   hc_ratio                                3643 non-null   float64
 4   headcount_international_povline         3643 non-null   float64
 5   total_shortfall_international_povline   3643 non-null   float64
 6   income_gap_ratio_international_povline  3643 non-null   float64
 7   international_poverty_gap               3643 non-null   float64
 8   $1_poverty_gap                          3643 non-null   float64
 9   decile1_avg                             3643 non-null   float64
 10  decile2_avg                             3643 non-null   float64
 

In [36]:
FilteredData.describe()

year     hc_ratio  headcount_international_povline  \
count  3643.000000  3643.000000                     3.643000e+03   
mean   2005.715893    12.138591                     1.544557e+07   
std       9.498677    18.918983                     7.161914e+07   
min    1967.000000     0.003111                     6.500000e+01   
25%    2000.000000     0.499448                     4.095900e+04   
50%    2007.000000     3.040325                     4.219720e+05   
75%    2013.000000    14.733494                     3.555498e+06   
max    2021.000000    96.871427                     9.106297e+08   

       total_shortfall_international_povline  \
count                           3.643000e+03   
mean                            1.013818e+07   
std                             5.516805e+07   
min                             4.957973e+01   
25%                             3.521472e+04   
50%                             2.879832e+05   
75%                             2.496510e+06   
max                             1.007303e+09   

       income_gap_ratio_international_povline  international_poverty_gap  \
count                             3643.000000                3643.000000   
mean                                41.365639                   4.481115   
std                                 20.375678                   8.182057   
min                                  9.670899                   0.001171   
25%                                 24.778685                   0.214716   
50%                                 37.090428                   0.894498   
75%                                 53.714336                   4.731381   
max                                100.415519                  64.096983   

       $1_poverty_gap  decile1_avg  decile2_avg  decile3_avg   gini_index  \
count     3643.000000  3643.000000  3643.000000  3643.000000  3643.000000   
mean         1.266590     5.069734     8.264604    10.382170     0.388367   
std          3.188800     5.839358     9.140140    11.129292     0.089323   
min          0.000001     0.000000     0.152899     0.238186     0.177920   
25%          0.045602     1.220718     2.018691     2.641458     0.321284   
50%          0.210813     2.298055     3.853281     5.104192     0.369068   
75%          0.863503     6.623261    11.340000    14.464495     0.447705   
max         41.989304    26.906183    40.561385    49.126067     0.657556   

       polarization  palma_index  
count   3643.000000  3643.000000  
mean       0.340420     2.019541  
std        0.103325     1.192297  
min        0.146643     0.596408  
25%        0.265295     1.233440  
50%        0.316390     1.567601  
75%        0.399454     2.407819  
max        0.815704     8.343586

In [37]:
FilteredData.columns

Index(['country', 'year', 'reporting_level', 'hc_ratio',
       'headcount_international_povline',
       'total_shortfall_international_povline',
       'income_gap_ratio_international_povline', 'international_poverty_gap',
       '$1_poverty_gap', 'decile1_avg', 'decile2_avg', 'decile3_avg',
       'gini_index', 'polarization', 'palma_index'],
      dtype='object')

In [38]:
FilteredData.shape


(3643, 15)

What it is: The Gini index is a measure of income inequality, ranging from 0 to 1:

    A Gini of 0 means perfect equality (everyone has the same income).
    A Gini of 1 means maximum inequality (one person has all the income, and everyone else has none).
 polarization:

    What It Does: The polarization index measures the extent to which a society is split into two distinct income groups (rich and poor), indicating the degree of social and economic division.
    Meaning: A higher polarization index means the society is becoming more divided, with fewer middle-income people and a larger gap between the rich and poor.
    Example: If income distribution becomes increasingly polarized, it means there is less of a middle class, and society is split between high-income and low-income groups.

In [39]:
#FilteredData[('decile','national')]=FilteredData[[('decile1_avg','national'),('decile2_avg','national'),('decile3_avg','national')]].mean(axis=1)
#FilteredData[('decile','urban')]=FilteredData[[('decile1_avg','urban'),('decile2_avg','urban'),('decile3_avg','urban')]].mean(axis=1)
#FilteredData[('decile','rural')]=FilteredData[[('decile1_avg','rural'),('decile2_avg','rural'),('decile3_avg','rural')]].mean(axis=1)
FilteredData['decile']=FilteredData[['decile1_avg','decile2_avg','decile3_avg']].mean(axis=1)
FilteredData.drop(['decile1_avg','decile2_avg','decile3_avg'],axis=1,inplace=True)
FilteredData

country  year reporting_level   hc_ratio  \
0      Albania  1996        national   0.920669   
1      Albania  2002        national   1.570843   
2      Albania  2005        national   0.860527   
4      Albania  2012        national   0.849754   
8      Albania  2016        national   6.681604   
...        ...   ...             ...        ...   
4871    Zambia  2006        national  64.698028   
4872    Zambia  2010        national  68.456606   
4873    Zambia  2015        national  61.352160   
4874  Zimbabwe  2011        national  21.580063   
4875  Zimbabwe  2017        national  34.206046   

      headcount_international_povline  total_shortfall_international_povline  \
0                             29167.0                           8.430020e+03   
1                             47927.0                           1.517408e+04   
2                             25915.0                           7.514215e+03   
4                             24646.0                           9.318746e+03   
8                            192170.0                           1.347542e+05   
...                               ...                                    ...   
4871                        7876026.0                           8.804439e+06   
4872                        9314196.0                           1.035156e+07   
4873                        9742337.0                           1.118811e+07   
4874                        2782603.0                           1.452058e+06   
4875                        4869778.0                           2.871835e+06   

      income_gap_ratio_international_povline  international_poverty_gap  \
0                                  15.211893                   0.140051   
1                                  16.663586                   0.261761   
2                                  15.260853                   0.131325   
4                                  19.900199                   0.169101   
8                                  36.906527                   2.465952   
...                                      ...                        ...   
4871                               51.994342                  33.639313   
4872                               51.691832                  35.386473   
4873                               53.413998                  32.770643   
4874                               24.271370                   5.237777   
4875                               27.429122                   9.382419   

      $1_poverty_gap  gini_index  polarization  palma_index    decile  
0           0.011726    0.270103      0.241293     0.928335  3.401783  
1           0.024816    0.317390      0.268982     1.215056  3.177819  
2           0.000230    0.305957      0.254529     1.142718  3.639047  
4           0.011959    0.289605      0.249988     1.041193  3.940452  
8           0.732899    0.385656      0.350360     1.696002  2.778895  
...              ...         ...           ...          ...       ...  
4871       12.755820    0.546175      0.548430     4.177633  0.580992  
4872       12.591384    0.556215      0.536149     4.278696  0.593405  
4873       13.713155    0.571361      0.613181     4.995829  0.550231  
4874        0.020411    0.431536      0.388356     2.207938  1.834881  
4875        0.253692    0.443371      0.416694     2.307359  1.488162  

[3643 rows x 13 columns]

In [40]:
FilteredData.isnull().sum()

country                                   0
year                                      0
reporting_level                           0
hc_ratio                                  0
headcount_international_povline           0
total_shortfall_international_povline     0
income_gap_ratio_international_povline    0
international_poverty_gap                 0
$1_poverty_gap                            0
gini_index                                0
polarization                              0
palma_index                               0
decile                                    0
dtype: int64

In [41]:
FilteredData.columns

Index(['country', 'year', 'reporting_level', 'hc_ratio',
       'headcount_international_povline',
       'total_shortfall_international_povline',
       'income_gap_ratio_international_povline', 'international_poverty_gap',
       '$1_poverty_gap', 'gini_index', 'polarization', 'palma_index',
       'decile'],
      dtype='object')

In [42]:
FilteredData.sort_values(by='reporting_level',inplace=True)
FilteredData.reset_index()

index      country  year reporting_level  hc_ratio  \
0         0      Albania  1996        national  0.920669   
1      3090  El Salvador  2009        national  7.419960   
2      3091  El Salvador  2010        national  6.345635   
3      3092  El Salvador  2011        national  5.340929   
4      3093  El Salvador  2012        national  4.637148   
...     ...          ...   ...             ...       ...   
3638   2478    Argentina  2007           urban  1.723668   
3639   2477    Argentina  2006           urban  2.118256   
3640   2476    Argentina  2005           urban  2.631179   
3641   2474    Argentina  2003           urban  5.279263   
3642   2819        China  2016           urban  0.241887   

      headcount_international_povline  total_shortfall_international_povline  \
0                             29167.0                           8.430020e+03   
1                            456897.0                           2.632790e+05   
2                            392406.0                           2.150010e+05   
3                            331702.0                           1.669509e+05   
4                            289262.0                           1.498491e+05   
...                               ...                                    ...   
3638                         618127.0                           5.117552e+05   
3639                         750699.0                           6.072329e+05   
3640                         921325.0                           6.745190e+05   
3641                        1803369.0                           1.428065e+06   
3642                        1904563.0                           1.066668e+06   

      income_gap_ratio_international_povline  international_poverty_gap  \
0                                  15.211893                   0.140051   
1                                  26.801526                   1.988661   
2                                  25.483933                   1.617116   
3                                  23.410043                   1.250314   
4                                  24.094853                   1.117315   
...                                      ...                        ...   
3638                               38.507569                   0.663743   
3639                               37.622787                   0.796947   
3640                               34.052017                   0.895969   
3641                               36.831974                   1.944457   
3642                               26.049264                   0.063010   

      $1_poverty_gap  gini_index  polarization  palma_index    decile  
0           0.011726    0.270103      0.241293     0.928335  3.401783  
1           0.152806    0.458251      0.390030     2.550027  2.957409  
2           0.142835    0.435141      0.391331     2.250383  3.068498  
3           0.070231    0.423184      0.368668     2.097612  3.207613  
4           0.067280    0.418293      0.355535     2.048324  3.389374  
...              ...         ...           ...          ...       ...  
3638        0.293898    0.461736      0.443909     2.639581  6.493650  
3639        0.259502    0.463461      0.446177     2.668039  6.193833  
3640        0.315964    0.477074      0.462944     2.877192  5.347521  
3641        0.679037    0.508972      0.512371     3.456020  4.006054  
3642        0.018087    0.361248      0.313856     1.521400  5.964138  

[3643 rows x 14 columns]

In [43]:
FilteredData.columns

Index(['country', 'year', 'reporting_level', 'hc_ratio',
       'headcount_international_povline',
       'total_shortfall_international_povline',
       'income_gap_ratio_international_povline', 'international_poverty_gap',
       '$1_poverty_gap', 'gini_index', 'polarization', 'palma_index',
       'decile'],
      dtype='object')

In [44]:
FilteredData['$1_poverty_gap'].describe()

count    3643.000000
mean        1.266590
std         3.188800
min         0.000001
25%         0.045602
50%         0.210813
75%         0.863503
max        41.989304
Name: $1_poverty_gap, dtype: float64

In [45]:
FilteredData['international_poverty_gap'].describe()

count    3643.000000
mean        4.481115
std         8.182057
min         0.001171
25%         0.214716
50%         0.894498
75%         4.731381
max        64.096983
Name: international_poverty_gap, dtype: float64

In [46]:
FilteredData['hc_ratio'].describe()

count    3643.000000
mean       12.138591
std        18.918983
min         0.003111
25%         0.499448
50%         3.040325
75%        14.733494
max        96.871427
Name: hc_ratio, dtype: float64